### PR TITLE
Weather

### PR DESCRIPTION
**Pull request by FrancoRosa**
**Dear TSEs,**
In this project I did the following:

- Used asynchronous functions to get data from API.
- Applied animation to be shown while the data is been requested.
- Added a switch to convert values and units.

Please give me some feedback, to improve this project.

### CODE REVIEW

File: src/js/maps.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+};
+
+export const getCityLocation = async () => {
+  const city = document.querySelector('.city-name').value;
+  const GoogleServer = 'https://maps.googleapis.com/maps/api/geocode/json?address={city}&key={geo_key}';
+  const GeoUrl = GoogleServer
+    .replace('{geo_key}', process.env.GOOGLE_GEO_API_KEY)
+    .replace('{city}', city)
+    .replace(' ', '%20');
+  const response = await fetch(GeoUrl, { method: 'GET' });
```
Comment (wathigo) [inline src/js/maps.js]: **REQUIRED**
* It has error handling for `promises` or `async/await` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/error_handling_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59622133/)).

File: src/js/weather.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+const events = eventMngr();
+
+const getWeatherURL = (lat, lon) => {
+  const WEATHER_SERVER = 'https://api.openweathermap.org/data/2.5/onecall?lat={lat}&lon={lon}&exclude=daily,hourly,minutely&appid={key}';
+  const url = WEATHER_SERVER.replace('{lat}', lat).replace('{lon}', lon).replace('{key}', process.env.OW_API_KEY);
+  return url;
+};
+
+const askWeather = async (lat, lon) => {
+  const response = await fetch(getWeatherURL(lat, lon), {
```
Comment (wathigo) [inline src/js/weather.js]: **REQUIRED**
* It has error handling for `promises` or `async/await` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/error_handling_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59622133/)).

**Review by wathigo — CHANGES_REQUESTED** [review]
### Changes Required.
Great job guys :clap:

**REQUIRED**
Let's add some error handling for for `promises` or `async/await` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/error_handling_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59622133/)).

**After making the changes, submit another request for review.**

**Happy Coding :computer: 
Simon Wathigo.**

**Comment by FrancoRosa** [comment]
I implemented the changes requested.
Now, the error handling gives better feedback.
image

**Review by uryelah — APPROVED** [review]
Hi **Franco Rosa Alagón**, I'm Sarah your reviewer for this PR 👩‍🚀 

Good job adding error handling to you app!

Go and rock that JS!  👨‍🎤   🎸  

-------

## **Your project is approved! 🎊 🥅 💛**

**You may close this Pull Request and merge the changes**


---------

    
    
**Keep being awesome devs and good luck with your next project!**

---------

Cheers and Happy coding!👏👏👏

Code review by **Sarah Uryelah Silva Chamorro**

(Feel free to contact me on slack by the handle of  **sarah** if you have any additional question)

**Pull request merged**